pygfx/renderers/wgpu/meshrender_phong.py

```diff
@@ -0,0 +1,315 @@
+
+    # Put it together!
+    return [
+        {
+            "suggested_render_mask": suggested_render_mask,
+            "render_shader": shader,
+            "primitive_topology": topology,
+            "cull_mode": cull_mode,
+            "indices": (range(n), range(n_instances)),
+            "index_buffer": index_buffer,
+            "vertex_buffers": vertex_buffers,
+            "bindings0": bindings,
+            "bindings1": bindings1,
+        }
+    ]
+
+
+class MeshPhongShader(WorldObjectShader):
+    def get_code(self):
+        return (
+            self.get_definitions()
+            + self.common_functions()
+            + mesh_vertex_shader
+            + lights
+            + bsdfs
+            + blinn_phong
+            + self.fragment_shader()
+        )
+
+    def fragment_shader(self):
+        return """
+
+        @stage(fragment)
+        fn fs_main(varyings: Varyings, @builtin(front_facing) is_front: bool) -> FragmentOutput {
+            $$ if color_mode == 'vertex'
+                let color_value = varyings.color;
+                let albeido = color_value.rgb;
+            $$ elif color_mode == 'map'
+                let color_value = sample_colormap(varyings.texcoord);
+                let albeido = color_value.rgb;  // no more colormap
+            $$ elif color_mode == 'normal'
+                let albeido = normalize(varyings.normal.xyz) * 0.5 + 0.5;
+                let color_value = vec4<f32>(albeido, 1.0);
+            $$ else
+                let color_value = u_material.color;
+                let albeido = color_value.rgb;
+            $$ endif
+
+            var i: i32 = 0;
+            var material: BlinnPhongMaterial;
+
+            material.diffuse_color = albeido;
+            material.specular_color = u_material.specular_color.rgb;
+            material.specular_shininess = u_material.shininess;
+
+            material.specular_strength = 1.0;  //TODO: Use specular_map if exists
+
+            var reflected_light: ReflectedLight = ReflectedLight(vec3<f32>(0.0), vec3<f32>(0.0), vec3<f32>(0.0), vec3<f32>(0.0));
+
+
+            $$ if num_dir_lights > 0 or num_point_lights > 0
+                let view_dir = select(
+                    normalize(u_stdinfo.cam_transform_inv[3].xyz - varyings.world_pos),
+                    ( u_stdinfo.cam_transform_inv * vec4<f32>(0.0, 0.0, 1.0, 0.0) ).xyz,
+                    is_orthographic()
+                );
+
+                var normal = varyings.normal;
+
+                if (u_material.flat_shading != 0 ) {
+                    let u = dpdx(varyings.world_pos);
+                    let v = dpdy(varyings.world_pos);
+                    normal = normalize(cross(u, v));
+                    normal = select(normal, -normal, (select(0, 1, is_front) + u_stdinfo.flipped_winding) == 1);  //?
+                }
+
+                // normal = select(-normal, normal, is_front);  // do we really need this?
```

Comment:
Do we really need this line of code? This line of code should cause the side facing away from the light to be illuminated also, which doesn't seem to be normal behavior?

The side facing the other direction should be illuminated by the light from the other direction. I don't think the light in one direction should illuminate both sides at the same time, so I commented it out for the time being.

The specific effect can be seen in my updated "validate" example. Now, the light in the opposite direction illuminates the opposite sides respectively.